Makefile

```diff
@@ -1,3 +1,227 @@
+AEA_SRC_DIR := aea
```

Comment:
Significant makefile update:
- Added and used constants 
- Ordered based on the popularity of usage.
- Explained all commands with comments.
- Updated those that do not work
- Used in tox (to avoid having two versions of a command that need to be updated and kept in sync).

Reply:
What about to move out benchmarks?
not even sure it works, not sure it's actually needed

Reply:
Let's keep it. Even if it won't be worked on for a while the checks and tests pass and should keep passing. Also I think it is important enough to pick up in not too distant future.